frontend/src/routes/project/[project_code]/+page.svelte

```diff
@@ -8,93 +8,141 @@
 	import t from '$lib/i18n';
 	import type { PageData } from './$types';
 	import DeleteModal from '$lib/components/modals/DeleteModal.svelte';
-	import { _deleteProjectUser, type ProjectUser } from './+page';
+	import {
+		_changeProjectDescription,
+		_changeProjectName,
+		_deleteProjectUser,
+		type ProjectUser,
+	} from './+page';
+	import EditableText from '$lib/components/EditableText.svelte';
+	import { Badge, BadgeList, TaggedBadge } from '$lib/components/Badges';
+	import { z } from 'zod';
+	import { user } from '$lib/user';
 
 	export let data: PageData;
 
 	$: project = data.project;
+	$: _project = project as NonNullable<typeof project>;
 
 	let deleteUserModal: DeleteModal;
-	let userToDelete: ProjectUser|undefined;
+	let userToDelete: ProjectUser | undefined;
 	async function deleteProjectUser(projectUser: ProjectUser) {
 		userToDelete = projectUser;
 		await deleteUserModal.prompt(async () => {
 			await _deleteProjectUser(project!.id, projectUser.User.id);
 		});
 	}
-</script>
 
-{#if project}
-	<div>
-		<p>
-			<span class="text-2xl">{project.name}</span>
-		</p>
+	const canManage = $user?.role == 'admin' || $user?.projects.find(project => project.code == project.code)?.role == 'Manager';
```

Comment:
this should probably be reactive (use $:) because the project comes from routed data it could load and swap out new data. Also the user role may automatically update in the future.